### PR TITLE
feat: adding formatValue to allow customizing the value format

### LINKED ISSUE #2040: SliderField renderValue function
### On which framework/platform would you like to see this feature implemented?

React

### Which UI component is this feature-request for?

Other

### Please describe your feature-request in detail.

I'd like the SliderField component to have a `renderValue` function where a user can customize the displayed value. My use case is I want to display from a value of 0.79 => 79%. 

### Please describe a solution you'd like.

Here is my current workaround solution:

```ts
import { ReactElement } from "react";
import { Placeholder, SliderField, SliderFieldProps } from "@aws-amplify/ui-react";
import { useController } from "react-hook-form";
import type { FieldValues } from "react-hook-form";
import { Box } from "../index.js";
import { BaseSmartInputProps } from "./baseProps.js";

export interface SmartSliderFieldProps<T>
  extends BaseSmartInputProps<T>,
    Omit<SliderFieldProps, "name"> {
      renderValue?: (v: string) => string;
    }

export function SmartSliderField<T extends FieldValues>(
  props: SmartSliderFieldProps<T>
): ReactElement {
  const {
    control,
    errorMessage,
    hasError,
    label,
    loading,
    name,
    renderValue,
    ...sliderFieldProps
  } = props;
  const {
    field: { ref, onChange, value },
    fieldState: { error, invalid },
  } = useController({ name, control });
  let Label: undefined | ReactElement;
  if (renderValue) {
    Label = <label data-testid="slider-label" className="amplify-label amplify-sliderfield__label">
      <span>{label}</span>
      <span>{renderValue(value)}</span>
    </label>
  }
  return loading ? (
    <Box css={{ display: "flex", flexDirection: "column", gap: "$2" }}>
      <label className="amplify-label">{label}</label>
      <Placeholder height={40} />
    </Box>
  ) : (
    <div>
      {Label}
      <SliderField
        labelHidden={Boolean(Label)}
        isValueHidden={Boolean(Label)}
        {...(sliderFieldProps as Omit<SliderFieldProps, "label" | "name">)}
        ref={ref}
        errorMessage={errorMessage || error?.message}
        hasError={hasError || invalid}
        name={name}
        label={label}
        onChange={onChange}
        value={value}
      />
    </div>
  );
}
```

### We love contributors! Is this something you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request.
- [ ] ⚠️ This feature might incur a breaking change.

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Adding a new `formatValue` prop to `SliderField` so that the value is allowed to be formatted

#### Issue #, if available

Fixes #2040 

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by calebpollman — COMMENTED** [review]
Looks like a pretty straightforward addition. Left some thoughts, lmk what you think

File: packages/react/src/primitives/SliderField/SliderField.tsx [hunk 1]
```diff
@@ -109,7 +114,9 @@ const SliderFieldPrimitive: Primitive<SliderFieldProps, typeof Root> = (
         visuallyHidden={labelHidden}
       >
         <View as="span">{label}</View>
-        {!isValueHidden ? <View as="span">{currentValue}</View> : null}
+        {!isValueHidden
+          ? formattedValue ?? <View as="span">{currentValue}</View>
+          : null}
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply packages/react/src/primitives/SliderField/SliderField.tsx:119]: That sounds like a good idea. Would like to hear more input here :0

**Review by calebpollman — APPROVED** [review]
LGTM 🚢 . Just left a request for an extra unit test

File: packages/react/src/primitives/types/sliderField.ts [hunk 2]
```diff
@@ -74,6 +75,12 @@ export interface SliderFieldProps extends TextInputFieldProps, ViewProps {
    */
   defaultValue?: number;
 
+  /**
+   * @description
```
Comment (0618) [inline packages/react/src/primitives/types/sliderField.ts:79]: We may want to update the props table by running `yarn docs build:props-tables`

File: packages/react/src/primitives/SliderField/SliderField.tsx [hunk 3]
```diff
@@ -77,6 +78,17 @@ const SliderFieldPrimitive: Primitive<SliderFieldProps, typeof Root> = (
     [onChange]
   );
 
+  const renderedValue = React.useMemo(() => {
```
Comment (joebuono) [inline packages/react/src/primitives/SliderField/SliderField.tsx:81]: Nice!

File: docs/src/pages/[platform]/components/sliderfield/react.mdx [hunk 4]
```diff
@@ -135,6 +136,19 @@ To add icons on either side of the SliderField, you may use the `outerStartCompo
   </ExampleCode>
 </Example>
 
+### Format value
+To format how the value looks like, you can pass in a render function to `formatValue` prop.
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/sliderfield/react.mdx:140]: "To format how the value is displayed..."

File: .changeset/red-pigs-rhyme.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+"@aws-amplify/ui-react": minor
+---
+
+feat: adding formatValue to allow customizing the value format
```
Comment (joebuono) [inline .changeset/red-pigs-rhyme.md:5]: Perhaps worth mentioning that this is for the SliderField specifically

**Review by joebuono — DISMISSED** [review]
LGTM! Just a few minor comments

File: packages/react/src/primitives/types/sliderField.ts [hunk 6]
```diff
@@ -74,6 +75,12 @@ export interface SliderFieldProps extends TextInputFieldProps, ViewProps {
...
+   * Uses to format how the value gets rendered
```
Comment (thaddmt) [inline packages/react/src/primitives/types/sliderField.ts:80]: Nit:

```suggestion
   * Use to format how the value gets rendered
```

File: packages/react/src/primitives/SliderField/__tests__/SliderField.test.tsx [hunk 7]
```diff
@@ -85,13 +86,43 @@ describe('SliderField: ', () => {
 
     it('should display value if isValueHidden is false', async () => {
       render(<SliderField defaultValue={10} label="slider" />);
-      const value = await screen.queryByText('10');
+      const value = screen.queryByText('10');
       expect(value).toBeInTheDocument();
     });
 
+    it('should display string formatted value if formatValue is provided', async () => {
```
Comment (thaddmt) [inline packages/react/src/primitives/SliderField/__tests__/SliderField.test.tsx:93]: Nit: should we have a test for null, undefined, or other unhappy case values?

**Review by thaddmt — APPROVED** [review]
LGTM, just small nits

File: packages/react/src/primitives/SliderField/__tests__/SliderField.test.tsx (hunk 7 above)
Comment (zchenwei) [reply packages/react/src/primitives/SliderField/__tests__/SliderField.test.tsx:93]: Let me add a test to cover these 2 nil values

Comment (zchenwei) [reply packages/react/src/primitives/SliderField/__tests__/SliderField.test.tsx:93]: The basic use test should kinda cover nil value, where by default the prop should be `undefined`

**Review by 0618 — APPROVED** [review]

**Review by joebuono — APPROVED** [review]
